iib/workers/tasks/build.py

```diff
@@ -747,19 +747,19 @@ def handle_add_request(
     if greenwave_config:
         gate_bundles(resolved_bundles, greenwave_config)
 
+    prebuild_info = _prepare_request_for_build(
```

Comment:
Instead of moving this function call to earlier, could you just move the logic in `_prepare_request_for_build` of determining the OCP version to a separate helper function that you call here?

I like the current flow where all validation occurs before the "Building the index image for the following arches: ..." state reason gets set.

Reply:
Hmm. It's quite odd that `_prepare_request_for_build` does set this state. But it's even worse if there's an intermediate state before the image actually gets built. I'll play with restructuring the code.

Reply:
Done. I pulled the state update out of the function.

Reply:
That's a clean approach. Thanks!